### PR TITLE
Make main() code consistent

### PR DESCRIPTION
**Pull request by rynowak**
# Description

This change updates each entry point of our microservices to the following in a more consistent way.

- Use cobra for argument parsing and help.
- Accept the configuration file using `--config-file` as a commandline option.
- Use a consistent pattern for choosing the ports used for various purposes.

The port changes only affect local development. In Kuberentes each pod can choose its own listening port without conflict.

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

## Contributor checklist
Please verify that the PR meets the following requirements, where applicable:

- [ ] An overview of proposed schema changes is included in a linked GitHub issue.
- [ ] A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
- [ ] If applicable, design document has been reviewed and approved by Radius maintainers/approvers.
- [ ] A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
- [ ] A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
- [ ] A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -1,126 +1,130 @@
...
-            "default": "init --full"
-        }
-    ]
-}
\ No newline at end of file
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  "version": "0.2.0",
+  "configurations": [
```
Comment (rynowak) [inline .vscode/launch.json:6]: This was an autoformat. I think @ytimocin updated this file during the addition of prettier, but it didn't get checked in. 

We intentionally `.gitignore` this file, so it requires some effort to change 😆

File: .vscode/launch.json [hunk 2]
```diff
@@ -1,126 +1,130 @@
...
+      "name": "Launch UCP",
+      "type": "go",
+      "request": "launch",
+      "mode": "auto",
+      "preLaunchTask": "Build Radius (all)",
+      "program": "${workspaceFolder}/cmd/ucpd/main.go",
+      "cwd": "${workspaceFolder}",
+      "args": [
+        "--config-file",
+        "${workspaceFolder}/cmd/ucpd/ucp-dev.yaml"
```
Comment (rynowak) [inline .vscode/launch.json:52]: This was changed, UCP now uses `--config-file` instead of an env-var.

File: cmd/applications-rp/cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+			client := hosting.NewAsyncValue[etcdclient.Client]()
+			options.Config.StorageProvider.ETCD.Client = client
+			options.Config.SecretProvider.ETCD.Client = client
+
+			hostingSvc = append(hostingSvc, data.NewEmbeddedETCDService(data.EmbeddedETCDServiceOptions{ClientConfigSink: client}))
+		}
+
+		builders, err := builders(options)
+		if err != nil {
+			log.Fatal(err) //nolint:forbidigo // this is OK inside the main function.
```
Comment (rynowak) [inline cmd/applications-rp/cmd/root.go:98]: Will remove this. We don't have a reason to use `log.Fatal` or similar constructs in this code.

**Review by kachawla — DISMISSED** [review]
Thanks for doing this!

File: cmd/applications-rp/cmd/root.go (hunk 3 above)
Comment (rynowak) [reply cmd/applications-rp/cmd/root.go:98]: I will do this now :). Sometimes I comment on my own PRs.

**Review by kachawla — APPROVED** [review]
:shipit:

**Pull request merged**